api/helpers/XForwardedForHelper.cs

```diff
@@ -1,11 +1,38 @@
-﻿namespace SS.Api.helpers
+﻿using System;
+
+namespace Scv.Api.Helpers
 {
     public static class XForwardedForHelper
     {
-        public static string BuildUrlString(string forwardedHost, string forwardedPort, string baseUrl)
+        public static string BuildUrlString(string forwardedHost, string forwardedPort, string baseUrl, string remainingPath = "", string query = "")
         {
-            var portComponent = string.IsNullOrEmpty(forwardedPort) || forwardedPort == "80" || forwardedPort == "443" ? "" : $":{forwardedPort}";
-            return $"https://{forwardedHost}{portComponent}{baseUrl}";
+            var sanitizedPath = baseUrl;
+            if (!string.IsNullOrEmpty(remainingPath))
+            {
+                sanitizedPath = string.Format("{0}/{1}", baseUrl.TrimEnd('/'), remainingPath.TrimStart('/'));
+            }
+
+            var uriBuilder = new UriBuilder
+            {
+                Scheme = "https",
+                Host = forwardedHost,
+                Path = sanitizedPath,
+                Query = query
+            };
+
+            var portComponent =
+                string.IsNullOrEmpty(forwardedPort) || forwardedPort == "80" || forwardedPort == "443"
```

Comment:
I added port 8080 here for scv to support running on Emerald.

Reply:
Purposely not doing that to keep things as similar to what's in `prod` `silver` now.